pandas/core/arrays/categorical.py

```diff
@@ -2003,7 +2007,18 @@ def __repr__(self) -> str:
 
     # ------------------------------------------------------------------
 
-    def __getitem__(self, key):
+    @overload
+    def __getitem__(self, key: int | np.integer) -> Any:
+        ...
+
+    @overload
+    def __getitem__(
+        self: CategoricalT,
+        key: slice | np.ndarray | list[int] | PositionalIndexerTuple,
```

Comment:
Would it be helpful to create a union for `slice | np.ndarray | list[int]`? There are a few places where we want to allow non-scalar/vector indices, e.g., for [take](https://github.com/pandas-dev/pandas/pull/43418).

I'm fine with using `list[int]`, some people might prefer `Sequence[int]` (if any Sequence is supported).

Reply:
`Sequence[int]` won't work because we don't want to support tuples here.

I'll create the other unions in `__typing__.py`